src/test/java/automationpractice/com/features/NewsletterFeatures.java

```diff
@@ -0,0 +1,47 @@
+package automationpractice.com.features;
+
+import automationpractice.com.pages.Footer;
+import automationpractice.com.pages.HomePage;
+import automationpractice.com.pages.LoginPage;
+import cucumber.api.PendingException;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.assertj.core.api.SoftAssertions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NewsletterFeatures {
+
+    final private String randomString = RandomStringUtils.randomAlphanumeric(13);
+    final private String randomEmail = String.format("%s@gmai.com", randomString);
+
+    private HomePage homePage;
+    private LoginPage loginPage;
+    private LoginPage email;
+    private Footer footer;
+
```

Comment:
1. Masz 2 razy LoginPage (z czego jeden nazywa się email) i nie używasz żadnego. Usuń to